lib/Parse/Lexer.cpp

```diff
@@ -1794,6 +1800,33 @@ static void validateMultilineIndents(const Token &Str,
                                   commonIndentation);
 }
 
+/// validateStringLiteralIsASCII:
+/// Diagnose contents of string literals that have non-printable-ASCII
+/// characters when compiling with -string-literals-must-be-ascii-only.
+static void validateStringLiteralIsASCII(const Lexer &Lexer, const Token &Str,
+                                         DiagnosticEngine *Diags) {
+  SmallVector<Lexer::StringSegment, 4> Segments;
+  Lexer::getStringLiteralSegments(Str, Segments, Diags);
+  for (auto &Seg : Segments) {
+    llvm::SmallString<256> Buf;
+    StringRef EncodedStr = Lexer.getEncodedStringSegment(Seg, Buf);
+
+    const char *BytesPtr = EncodedStr.begin();
+    while (BytesPtr < EncodedStr.end()) {
+      char CurChar = *BytesPtr++;
+
+      // Printable ASCII characters
+      if (CurChar >= 0x20 && CurChar <= 0x7e) continue;
```

Comment:
The check taking place here isn't whether or not the string is ASCII; it apparently has to be _printable_ ASCII plus `TAB`, `CR` and `LF`.

IMO if we're testing for ASCII-only strings, it should allow anything up to and including `DEL` (`0x7f`).

It'd be quite reasonable to have a Swift program running in ASCII-only mode that emits VT102 control codes, for instance; this would effectively prohibit that.

Reply:
Makes sense, let’s do what you’re suggesting.